app/models/user.rb

```diff
@@ -390,6 +390,14 @@ def is_archivist?
     has_role?(:archivist)
   end
 
+  # Is this user an authorized official?
+  def official
+    roles.each do |role|
+      return true if role.name == "official"
+    end
+    false
```

Comment:
I think it'd be cleaner to rewrite this using [any?](https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-any-3F).

(You could have continued using `has_role?` here and instead modified the [definition](https://github.com/otwcode/otwarchive/blob/aa9204948ca8c5104ef00f965f46dd4276f2b549/config/initializers/gem-plugin_config/permit_yo_config.rb#L23-L28) of `has_role?`, possibly with a call to `if roles.loaded?` if you wanted to keep the existing behavior when the roles aren't loaded. But I don't think there are any more examples of N+1 queries involving roles in the current codebase, so I'm not sure if that kind of future-proofing is necessary.)